### ISSUE TITLE
Duplicate styles/padding for links in Dropdown

### ISSUE THREAD

**Issue by ocean90** [primer/components#902]
**Describe the bug**

The current styles for `Dropdown.Item` prevents having the same padding for text only items and items with links.

https://github.com/primer/components/blob/7b8de76ee1b0bd5aaf99d9514b9fea870bf5464e/src/Dropdown.js#L93-L109

**Expected behavior and screenshots**
I'd expect to not have double padding for links inside of a dropdown item.

```
<Dropdown.Item>Text only</Dropdown.Item>
<Dropdown.Item><Link href="#">Sign out</Link></Dropdown.Item>
```

<img width="211" alt="Bildschirmfoto 2020-11-08 um 14 12 40" src="https://user-images.githubusercontent.com/617637/98465925-9adaaf00-21cc-11eb-92d1-08b445193756.png">

Also tried with `as="a"`:

```
<Dropdown.Item>No link</Dropdown.Item>
<Dropdown.Item as="a" href="#">Sign out</Dropdown.Item>
```

<img width="223" alt="Bildschirmfoto 2020-11-08 um 14 08 39" src="https://user-images.githubusercontent.com/617637/98465945-b04fd900-21cc-11eb-8fcf-f58d35103afd.png">

Happy to provide a PR with a fix but I'm not sure which variant is expected to work.

**Comment by ocean90** [comment]
Looking at https://primer.style/css/components/dropdown and it only has examples for links. At least the double padding doesn't seem to be intended.

@emplums Do you have any thoughts on this?

**Comment by emplums** [comment]
Good catch, this is a bug!

**Comment by emplums** [comment]
Fixed in 22.0.0 🎉
